packages/integrations/markdoc/test/render.test.js

```diff
@@ -125,6 +147,26 @@ function renderComponentsChecks(html) {
 	expect(pre.className).to.equal('astro-code github-dark');
 }
 
+/** @param {string} html */
+function renderIndentedComponentsChecks(html) {
+	const { document } = parseHTML(html);
+	const h2 = document.querySelector('h2');
+	expect(h2.textContent).to.equal('Post with indented components');
+
+	// Renders custom shortcode components
+	const marquees = document.querySelectorAll('marquee');
+	expect(marquees.length).to.equal(2);
+
+	// Renders indented h3
+	const h3 = document.querySelector('h3');
+	expect(h3.textContent).to.equal('I am an h3!');
```

Comment:
How do you know if this is indented? Should we check for spaces/tabs somehow?

Reply:
Unfortunately this doesn't answer my question. The comment in the test says "we check if text is indented" , but still I can't understand when a text should not be indented. Sorry for those dumb questions , I have little context about markdoc

Reply:
@ematipico No worries ! I can see how these comments are a bit confusing - I've updated them.

The `allowIndentation` option is purely to enable better readability of source code and it will not affect the rendered HTML. Check out Markdoc's official documentation on indentation https://markdoc.dev/docs/faq#indentation

- without `allowIndentation` enabled:
  ```
  {% foo %}
  {% bar %}
  This is content inside of nested tags
  {% /bar %}
  {% /foo %}
  ```

- with `allowIndentation` enabled (easier to read, level of depth is clear)

  ```
  {% foo %}
    {% bar %}
      This is content inside of nested tags
    {% /bar %}
  {% /foo %}
  ```

The rendered HTML is exactly the same in both examples. The second example will fail to render if the `allowIndentation` option is not enabled.

I hope this helps!

Reply:
Do you think `ignoreIndentation: true` might be better here?

It makes sense for it to be called `allowIndentation` in markdoc's tokenizer options. It might be better to hint at its overall effect with the name of the top-level option.

Reply:
Yeah `ignoreIndentation` sounds good !